sw/device/lib/crypto/drivers/kmac_test.c

```diff
@@ -76,42 +80,105 @@ bool test_main(void) {
   // Below we repeat some of the test above, but through higher-level
   // cryptolib API. The gist is to test if glueing is done correctly.
   for (i = 0; i < nist_kmac_nr_of_vectors; i++) {
+    LOG_INFO("Testing cryptolib API (glueing) #%d", i);
     current_test_vector = nist_kmac_vectors[i];
 
-    crypto_const_uint8_buf_t input_buf;
-    input_buf.data = current_test_vector->input_str;
-    input_buf.len = current_test_vector->input_str_len;
+    hash_mode_t hash_mode;
+    xof_mode_t xof_mode;
+    mac_mode_t mac_mode;
 
-    crypto_uint8_buf_t digest_buf;
-    digest_buf.data = digest;
-    digest_buf.len = current_test_vector->digest_len;
+    // Max key size is 512 bits
+    uint32_t keyblob[2 * 512 / 32];
```

Comment:
nit: explain the `2 *` also.
```suggestion
    // Max key size is 512 bits and keys are in two shares.
    uint32_t keyblob[2 * 512 / 32];
```